lib/rbac/filterer.rb

```diff
@@ -389,19 +420,15 @@ def add_joins(klass, scope, includes)
       return scope unless includes
       includes = Array(includes) unless includes.kind_of?(Enumerable)
       includes.each do |association, value|
-        if table_include?(klass, association)
+        reflection = klass.reflect_on_association(association)
+        if reflection && !reflection.polymorphic?
           scope = value ? scope.left_outer_joins(association => value) : scope.left_outer_joins(association)
+          scope = scope.distinct if reflection.try(:collection?)
         end
       end
       scope
     end
 
-    # this is a reference to a non polymorphic table
-    def table_include?(target_klass, association)
```

Comment:
This through me off for a bit, but why did you move this method into `add_joins`?  I guess it is one less extra method in the class, which avoids others potentially using it for unintended purposes.  Just wasn't clear.

Reply:
Oh, I see.  You are using `reflection` now inside the if.

Of note to others:  `table_include?` was added here:  https://github.com/ManageIQ/manageiq/pull/18543

And there is a 99% chance that @kbrock is the only one to know of it's existance.